ingest/workflow/snakemake_rules/nextclade.smk

```diff
@@ -56,15 +56,31 @@ rule join_metadata_clades:
     input:
         nextclade="data/nextclade.tsv",
         metadata="data/metadata_raw.tsv",
+        nextclade_field_map="source-data/nextclade-field-map.tsv",
     output:
-        "data/metadata.tsv",
+        metadata="data/metadata.tsv",
     params:
         id_field=config["transform"]["id_field"],
     shell:
         """
-        python3 bin/join-metadata-and-clades.py \
-                --id-field {params.id_field} \
-                --metadata {input.metadata} \
-                --nextclade {input.nextclade} \
-                -o {output}
+        csvtk -tl rename2 \
+            -F \
+            -f '*' \
+            -p '(.+)' \
+            -r '{{kv}}' \
+            -k {input.nextclade_field_map} \
+            {input.nextclade} \
+            > results/nextclade_renamed.tsv
+
+        export APPEND_FIELDS=`awk 'NR>1 {{print $2}}' {input.nextclade_field_map} | grep -v -e "index" -e "seqName" | tr '\n' ',' | sed 's/,\$//g'`
+
+        cat {input.metadata} | tr '\r' '\n' | grep -v "^$" \
+        | tsv-join -H \
+            --filter-file results/nextclade_renamed.tsv \
+            --key-fields seqName \
+            --data-fields {params.id_field} \
+            --append-fields $APPEND_FIELDS \
+            --allow-duplicate-keys \
```

Comment:
Curious why we need to allow duplicates in the join? The `seqName` should be the GenBank accession, which should all be unique.

Reply:
Dropped in https://github.com/nextstrain/monkeypox/pull/207/commits/0ff67dfdd4a967f055c22923386de7e05aca1a33
I use a similar `tsv-join` in an unrelated pipeline where my keys are duplicated 😓 but you are correct, seqName here should be unique

Reply:
(Dropped in c21e9f80797bca1e8c4462328cc23bbfd76f0b8a)